azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -827,6 +818,71 @@ def _begin_delete_secret(self, keyvault_name: str, secret_name: str):
                 raise
 
         # Failed to confirm deletion after retries
-        raise TimeoutError(
-            f"Failed to delete secret '{secret_name}' within {SECRET_DELETE_MAX_RETRIES} retries."
-        )
+        raise TimeoutError(f"Failed to delete secret '{secret_name}' within {SECRET_DELETE_MAX_RETRIES} retries.")
+
+    def _extract_cert_content(
+        self,
+        public_key_file: str,
+    ) -> Tuple[str, str]:
+        from cryptography import x509
+        from cryptography.x509.oid import NameOID, ExtensionOID
+
+        der_data = read_file_content(file_path=public_key_file, read_as_binary=True)
+        certificate = decode_der_certificate(der_data)
+
```

Comment:
What do you think about a later iteration validation for cert expiry?

Reply:
good idea, since we already decoded the cert content, will add TODO